### ISSUE TITLE
question

### ISSUE THREAD

**Issue by BannerBomb** [John10v10/MKTrackMaker#2]
can you upload the compiled program or windows please cause I don't have any windows computers since they all crashed. I am stuck with ubuntu but I can run a .exe using wine on ubuntu, so if you could upload the compiled windows program that would be nice.

**Comment by John10v10** [comment]
@BannerBomb, I have an .exe build uploaded. But the program overall is still incomplete and useless. Here's the build: MKTrackMaker/MarioKartTrackMaker/MKTMbuild.zip
 - John10v10

**Comment by BannerBomb** [comment]
honestly the tool could be very useful to modders, I am sure everyone would like a tool like this, if you decide to ever finish it.

**Comment by BannerBomb** [comment]
@John10v10 do you think you could add a Makefile to the project so I can modify the project on my computer and be able to build it? The Makefile should be for linux based systems to use make to build it. Or if you can't make a makefile for it then can you add something else to allow me to build the project on linux. Thanks. and sorry for posting so much.

**Comment by John10v10** [comment]
That's awesome, dude. Will you be able to do it again if/when I ever release V1 of the program?

**Comment by BannerBomb** [comment]
Yes, I will be able to. All I will need to change is the packages hosted on one of my github repos, the `build.sh` file should never have to be changed since the only one thing needing to be changed is on my github. Cause what the `build.sh` does is downloads the packages from my github repo and puts them in the correct folders if there is no `packages` folder detected then it changes some things in the .vcproj file so it can easily locate the packages folder and it also installs the required packages to linux **if** they are not already installed then it builds the project using the `.sln` file. So anything that changes in the project won't affect the `build.sh` file. Just if you want to you can change the github link in the `build.sh` file to your own if you want to host the packages yourself see my repo if you want to see how I zipped the files up. Otherwise leave the link and I will update the packages when ever you update this project.

**Comment by BannerBomb** [comment]
Also I though I should show these so you know what basically happens during and after compiling.
[https://imgur.com/F69jjuB](Before Compiling Root Folder)
[https://imgur.com/a/aePWZxY](Before Compiling Project Folder)
[https://imgur.com/a/HGPxFIc](This is where you should have the build.sh file located at)
[https://imgur.com/a/Yhab9VR](Compiling the project)
[https://imgur.com/a/6MJxiaw](Installing required packages failed)
If you see this error for the image above (Installing failed) there is an easy fix for it.
1. Open Software & Updates
2. Under the `Ubuntu software` tab, `Downloadable from the internet` section check all of the boxes except the `Source Code` checkbox.
3. Press `close`
4. Press `reload`
5. Try running `./build.sh` again.
**If you get Permission denied or something similar to that when running ./build.sh, it is because you did not run `chmod +x build.sh` on the file.**
[https://imgur.com/a/3YLGBGB](To Fix Installation Error)
[https://imgur.com/a/p8dMjaB](After Compiling Root Folder)
[https://imgur.com/a/aAjxIt6](After Compiling Project Folder)
[https://imgur.com/a/D1ryvyK](What the Build folder should look like after compiling)

**Added these to try to help explain the steps of using the build.sh file and before/after compiling the program what you should see added to the folders after compiling. (Project will look different than the ones in these images if/when you add more to the project of course)**
